framework/include/utils/RankTwoTensor.h

```diff
@@ -502,6 +533,9 @@ class RankTwoTensorTempl : public TensorValue<T>
   /// RankTwoTensorTempl<T> from outer product of vectors
   void vectorOuterProduct(const TypeVector<T> &, const TypeVector<T> &);
 
+  /// RankTwoTensorTempl<T> from outer product of a vector with itself
+  static RankTwoTensorTempl<T> vectorSelfOuterProduct(const TypeVector<T> &);
```

Comment:
This signature looks weird to me. It seems like you are not filling values in place, instead a new RankTwoTensor is returned?

Reply:
yes, this is a "named constructor" idiom.